tests/ztimer_periodic/main.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2020 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2020 Freie Universität Berlin
+ *               2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     test
+ * @{
+ *
+ * @file
+ * @brief       ztimer periodic test application
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+
+#include "ztimer.h"
+#include "ztimer/periodic.h"
+#include "fmt.h"
+#include "mutex.h"
+
+static mutex_t _mutex = MUTEX_INIT_LOCKED;
+
+#define MAX_OFFSET 2
+#define N 5
+#define INTERVAL 100LU
+
+static volatile uint32_t _times[N];
```

Comment:
If I recall we want to tart avoiding volatile for these cases.

Reply:
Good catch!

Reply:
so I use a volatile uint array?

Reply:
It should work with plain `uint32_t` here, as the `mutex_lock()` call prior to accessing the variables from within `main()` should have all the neseccary memory barriers (due to its use of `irq_disable()` and `irq_restore()`). One would need to compile with `LTO=1` to test this, as external function calls always are treated as memory barriers. (The compiler cannot know for sure if the function does not access `_times`, unless the implementation is visible. This would be the case with `LTO=1`.)

I personally would just go for C11 atomics and not having to argue why it is safe. But feel free to go with plain `uint32_t` instead.